### PR TITLE
Update documentation to generate result figures automatically

### PR DESCRIPTION
**Pull request by lin-yang-ly**
The python scripts and documentations of V&V cases are updated to generate figures automatically on documentation.

(Refs #168)

### CODE REVIEW

**Comment by lin-yang-ly** [comment]
The PR is ready for review! @simopier Could you add @chaibhave as a reviewer? Thank you!

**Comment by simopier** [comment]
You are getting the following errors in Apptainer:
```
ver-1g.ver-1g_comparison: Working Directory: /tmp/tmap8test.WvHFXB/tests/tmap8/tests/tests/ver-1g
ver-1g.ver-1g_comparison: Running command: python3 comparison_ver-1g.py
ver-1g.ver-1g_comparison: Traceback (most recent call last):
ver-1g.ver-1g_comparison:   File "/tmp/tmap8test.WvHFXB/tests/tmap8/tests/tests/ver-1g/comparison_ver-1g.py", line 20, in <module>
ver-1g.ver-1g_comparison:     tmap_sol = pd.read_csv(os.path.join(git.Repo('.',search_parent_directories=True).working_tree_dir, "test/tests/ver-1g/gold/diff_conc_TMAP4_out.csv"))
ver-1g.ver-1g_comparison:   File "/opt/miniforge3/envs/moose/lib/python3.10/site-packages/git/repo/base.py", line 289, in __init__
ver-1g.ver-1g_comparison:     raise InvalidGitRepositoryError(epath)
ver-1g.ver-1g_comparison: git.exc.InvalidGitRepositoryError: /tmp/tmap8test.WvHFXB/tests/tmap8/tests/tests/ver-1g
ver-1g.ver-1g_comparison: 
ver-1g.ver-1g_comparison: ################################################################################
ver-1g.ver-1g_comparison: Tester failed, reason: CODE 1
ver-1g.ver-1g_comparison: 
[1.146s]     FAIL ver-1g.ver-1g_comparison FAILED (CODE 1) [previous results: CODE 1]


Final Test Results:
--------------------------------------------------------------------------------------------------------------
[0.840s]       OK ver-1fa.ver-1fa [re-running]
[1.376s]       OK ver-1fd.ver-1fd [re-running]
[0.000s]       OK ver-1fd.ver-1fd_csv [re-running]
[2.590s]       OK ver-1gc.ver-1gc [re-running]
[3.067s]       OK ver-1ja.ver-1ja_csvdiff [re-running]
[0.000s]       OK ver-1gc.ver-1gc_csv [re-running]
[3.652s]       OK ver-1ha.ver-1ha_csv [re-running]
[3.780s]       OK ver-1fc.ver-1fc [re-running]
[3.801s]       OK ver-1g.binary_reaction_equal_concentrations [re-running]
[0.000s]       OK ver-1fc.ver-1fc_csvdiff_transient [re-running]
[0.000s]       OK ver-1fc.ver-1fc_csvdiff_steady_state [re-running]
[2.716s]       OK ver-1g.binary_reaction_diff_concentrations_TMAP4 [re-running]
[9.566s]       OK ver-1c.ver-1c [re-running]
[0.000s]       OK ver-1c.ver-1c_csvdiff [re-running]
[3.198s]       OK ver-1g.binary_reaction_diff_concentrations_TMAP7 [re-running]
[1.450s]     FAIL ver-1fa.ver-1fa_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.082s]     FAIL ver-1fd.ver-1fd_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.168s]     FAIL ver-1ja.ver-1ja_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.335s]     FAIL ver-1gc.ver-1gc_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.237s]     FAIL ver-1ha.ver-1ha_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.124s]     FAIL ver-1fc.ver-1fc_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.078s]     FAIL ver-1c.ver-1c_comparison FAILED (CODE 1) [previous results: CODE 1]
[1.146s]     FAIL ver-1g.ver-1g_comparison FAILED (CODE 1) [previous results: CODE 1]

File: doc/content/verification_and_validation/val-2b.md [hunk 1]
```diff
@@ -43,10 +43,15 @@ The model applies 13.3 kPa of D$_2$ for 50 hours and 15 seconds followed by evac
...
+       caption=Comparison of TMAP8 calculation with the experimental data
 
-!media figures/val-2b_comparison.png
-    style=width:50%;margin-bottom:2%
-    id=val-2b_comparison
-    caption=Comparison of TMAP8 calculation with the experimental data
+## Input files
+
+!style halign=left
+The input file for this case can be found at [/val-2b.i], which is also used as test in TMAP8 at [/val-2b/tests].
```
Comment (simopier) [inline doc/content/verification_and_validation/val-2b.md:55]: When you create branches, make sure to create separate branches based on devel rather than other branches. Otherwise, you'll create conflicts like this one.

File: doc/content/verification_and_validation/ver-1a.md [hunk 2]
```diff
@@ -142,8 +142,9 @@ Again, be aware of the typos in [!cite](ambrosek2008verification) and the differ
        id=ver-1a_comparison_analytical_TMAP7_flux
        caption=Comparison of TMAP8 calculation with the analytical solution for the release fraction from [!cite](ambrosek2008verification).
 
-## Comments on verification input file versus test file
+## Input files
 
+!style halign=left
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1a.md]: Same here regarding conflicts

File: test/tests/ver-1ja/tests [hunk 3]
```diff
@@ -13,6 +13,7 @@
     command = 'python3 comparison_ver-1ja.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1ja, which models decay of tritium and associated growth of helium in a diffusion segment.'
     prereq = ver-1ja_csvdiff
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1ja/tests:16]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1ha/tests [hunk 4]
```diff
@@ -13,6 +13,7 @@
       command = 'python3 comparison_ver-1ha.py'
       requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a convective outflow problem involving three enclosures and calculate the pressure and concentration of the gas in the second and third enclosure.'
       prereq = 'ver-1ha_csv'
-      required_python_packages = 'csv matplotlib numpy pandas scipy os math'
+      heavy = true
```
Comment (simopier) [inline test/tests/ver-1ha/tests:16]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1gc/tests [hunk 5]
```diff
@@ -22,6 +22,7 @@
     command = 'python3 comparison_ver-1gc.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a series of chemical reactions involving three species and calculate the concentrations of each species as a function of time and to generate CSV data for use in comparisons with the analytic solution over time.'
     prereq = 'ver-1gc_csv'
-    required_python_packages = 'csv matplotlib numpy pandas scipy os math'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1gc/tests:25]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1g/tests [hunk 6]
```diff
@@ -51,6 +51,7 @@
     command = 'python3 comparison_ver-1g.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a chemical reaction between two species with same or different concentrations, using the initial conditions from both TMAP4 and TMAP7 cases.'
     prereq = 'binary_reaction_equal_concentrations binary_reaction_diff_concentrations_TMAP4 binary_reaction_diff_concentrations_TMAP7'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1g/tests:54]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1fd/tests [hunk 7]
```diff
@@ -20,7 +20,8 @@
     type = RunCommand
     command = 'python3 comparison_ver-1fd.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1fd, modeling convective heating.'
-    required_python_packages = 'csv matplotlib scipy numpy pandas os math'
+    required_python_packages = 'matplotlib scipy numpy pandas os git'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1fd/tests:24]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1fc/tests [hunk 8]
```diff
@@ -37,6 +37,7 @@
     command = 'python3 comparison_ver-1fc.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution, ABAQUS data, and simulated solution of verification case 1fc, modeling conduction in a composite structure with constant surface temperatures.'
     prereq = 'ver-1fc_csvdiff_transient ver-1fc_csvdiff_steady_state'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1fc/tests:40]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1fa/tests [hunk 9]
```diff
@@ -21,6 +21,7 @@
     command = 'python3 comparison_ver-1fa.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1fa, to model heat conduction in a slab that has heat generation.'
     prereq = ver-1fa
-    required_python_packages = 'matplotlib numpy pandas os git'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1fa/tests:24]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1c/tests [hunk 10]
```diff
@@ -20,6 +20,7 @@
     command = 'python3 comparison_ver-1c.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1c, modeling species permeation into an unloaded portion of a slab from a pre-loaded portion for both the TMAP4 and TMAP7 verification cases.'
     prereq = 'ver-1c_csvdiff'
-    required_python_packages = 'matplotlib numpy pandas os git'
+    heavy = true
```
Comment (simopier) [inline test/tests/ver-1c/tests:23]:
```suggestion
```
Why do you need `heavy = true` here?

File: test/tests/ver-1b/comparison_ver-1b.py [hunk 11]
```diff
@@ -83,7 +88,7 @@
 D = 1
 analytical_flux = C_o * np.sqrt(D/(np.pi * analytical_time)) * \
                   np.exp( x / (2 * np.sqrt(D * analytical_time)))
-ax.plot(tmap_time,tmap_flux,label=r"TMAP8",c='tab:gray')
+ax.plot(tmap_time[1:],tmap_flux[1:],label=r"TMAP8",c='tab:gray')
```
Comment (simopier) [inline test/tests/ver-1b/comparison_ver-1b.py:91]: Is the [1:] really needed? Why?

File: doc/content/verification_and_validation/ver-1gc.md [hunk 12]
```diff
@@ -53,12 +53,18 @@ The time evolution of the species concentrations from the analytical solution is
...
+!media comparison_ver-1gc.py
+       image_name=ver-1gc_comparison_diff_conc.png
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1gc_comparison_diff_conc
+       caption=Comparison of partial pressures of species in a series reaction predicted by TMAP8 and provided by the analytical solution. The RMSPE is the root mean square percent error between the analytical solution and TMAP8 predictions.
+
+## Input files
+
+!style halign=left
+The input file for this case can be found at [/ver-1gc.i], which is also used as test in TMAP8 at [/ver-1gc/tests].
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1gc.md:67]: Conflict?

File: doc/content/verification_and_validation/ver-1gc.md [hunk 13]
```diff
@@ -53,12 +53,18 @@ The time evolution of the species concentrations from the analytical solution is
 
 ## Results and comparison against analytical solution
 
-The comparison of TMAP8 results against the analytical solution is shown in [ver-1gc_comparison_diff_conc]. The match between TMAP8's predictions and the analytical solution is satisfactory, with root mean square percentage errors (RMSPE) of RMSPE = 0.54 % for species $A$, RMSPE = 0.36 % for species $B$, and RMSPE = 0.04 % for species $C$. The larger values of RMSPE for species $A$ and $B$ are due to the small average values of these concentrations over time, and do not reflect on the accuracy of the TMAP8 solve itself.
+The comparison of TMAP8 results against the analytical solution is shown in [ver-1gc_comparison_diff_conc]. The match between TMAP8's predictions and the analytical solution is satisfactory, with root mean square percentage errors (RMSPE) of RMSPE = 4.78 % for species $A$, RMSPE = 3.20 % for species $B$, and RMSPE = 0.38 % for species $C$. The larger values of RMSPE for species $A$ and $B$ are due to the small average values of these concentrations over time, and do not reflect on the accuracy of the TMAP8 solve itself.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1gc.md:56]: I'm surprised the RMSPE is increasing so much (x10) since the documentation was first created!
Was it wrong in the first place?

File: doc/content/verification_and_validation/ver-1g.md [hunk 14]
```diff
@@ -44,15 +44,22 @@ where $P_{i_0}$ is the initial pressure, $N_a$ is Avogardro's constant, $R$ is t
...
+!media comparison_ver-1g.py
+       image_name=ver-1g_comparison_diff_conc.png
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1g_comparison_diff_conc
+       caption=Comparison of concentration of AB as a function of time calculated through TMAP8 and analytically for the case when A and B have different concentrations. The plot shows comparisons for the initial conditions specified in both TMAP4 (case (b)) and TMAP7 (case (c)).
+
+## Input files
+
+!style halign=left
+The input file for these cases can be found at [/ver-1g.i], which is also used as test in TMAP8 at [/ver-1g/tests]. The initial conditions of three cases (a), (b), and (c) can be found at [/equal_conc.i], [/diff_conc_TMAP4.i], and [/diff_conc_TMAP7.i].
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1g.md:62]: conflict?

File: doc/content/verification_and_validation/ver-1fd.md [hunk 15]
```diff
@@ -25,12 +25,18 @@ In [!cite](ambrosek2008verification), the value of $k = 801$ W/m/K is provided,
 
 The comparison between TMAP8 predictions and the analytical solution is performed at depth $x = 5$ cm.
 These results are shown in [ver-1fd_comparison_convective_heating].
-They show great agreement between TMAP8 and the analytical solution with a root mean square percentage error of RMSPE = 0.04 %.
+They show great agreement between TMAP8 and the analytical solution with a root mean square percentage error of RMSPE = 0.29 %.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1fd.md:28]: How is the RMSPE increased? Was the number wrong in the first place?

File: doc/content/verification_and_validation/ver-1fc.md [hunk 16]
```diff
@@ -40,13 +40,14 @@ profile of A and B can therefore be found through linear interpolation.
 
 With TMAP8, the steady state solution can be obtained in different ways: by using the [steady state solve](source/executioners/Steady.md) or by running a [transient simulation](source/executioners/Transient.md) until steady state is reached.
 [!cite](ambrosek2008verification) indicates that the steady state solution was obtained by running the transient solution until $t=10,000$ s, which is what is reproduced with TMAP8 here.
-TMAP8 predictions were found to be identical to the analytical solution with a root mean square percentage error (RMSPE) of 0.01 %, as shown in
+TMAP8 predictions were found to be identical to the analytical solution with a root mean square percentage error (RMSPE) of 0.23 %, as shown in
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1fc.md:43]: Was that number incorrect before? I am surprised to see an > 10 fold increase in several of these cases.

File: doc/content/verification_and_validation/ver-1b.md [hunk 17]
```diff
@@ -47,11 +49,12 @@ taken as unity, and the distance, x, was taken as 0.5 in this comparison.
 TMAP8 initially under predicts but the results match well subsequently. Comparison
 results are shown in []
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1b.md:50]:
```suggestion
results are shown in [ver-1b_comparison_flux].
```

Comment (simopier) [inline doc/content/verification_and_validation/ver-1b.md:50]: Can you please fix the reference in the above paragraph? The equation and table numbers are not updated. 
Maybe remove the table cross-reference since the documentation does not have a table.

File: doc/content/verification_and_validation/ver-1c.md [hunk 18]
```diff
@@ -23,17 +23,24 @@ TMAP4 and TMAP7 verification cases are slightly different: TMAP4 verifies the mo
...
+       image_name=ver-1c_comparison_time_TMAP7.png
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1c_comparison_time_TMAP7
+       caption=Comparison of concentration as a function of time at x\=0.25 m, 10 m, and 12 m
+       calculated with TMAP8 and analytically (TMAP7 cases)
+
+## Input files
+
+!style halign=left
+The input file for this case can be found at [/ver-1c.i], which is also used as test in TMAP8 at [/ver-1c/tests].
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1c.md:43]: The conflicts have not been resolved properly - the documentation pages now have two `## input files` section. Please remove the one added in this PR.

**Review by simopier — CHANGES_REQUESTED** [review]
Good work! This is going to be extremely valuable to keep the documentation updated.  Plus, it seems like we had incorrect RMSPE values in some of these pages - good catches!
However, I have some suggestions and fixes. Most important one: conflicts were not properly resolved. Many documentation pages now have two `## Input files` section. Please remove the one added in this PR.

File: doc/content/verification_and_validation/val-2b.md (hunk 1 above)
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/val-2b.md:55]: Because the PR of adding input section and the PR of updating figures automatically change the same documentation files, I created the current one based on the one adding input section to decrease the extra workload during merging. However, it seems that I add more workload here! I will create branch based on devel in future.

File: test/tests/ver-1ja/tests (hunk 3 above)
Comment (lin-yang-ly) [reply test/tests/ver-1ja/tests:16]: Because Apptainer test has the error once run the python script for the verification cases
`python3 comparison_ver-1dd.py`
The error doesn't appear on local machine. The cases without the error on Apptainer test environment use `heavy = true` to skip the running. So I add it here to avoid the issue on Apptainer test environment. I also discussed with @chaibhave about the issue. One possible reason is the git package in Apptainer test environment need to be updated.

File: test/tests/ver-1b/comparison_ver-1b.py (hunk 11 above)
Comment (lin-yang-ly) [reply test/tests/ver-1b/comparison_ver-1b.py:91]: In this case, the first data point in tmap_flux is zero and the first one in analytical_flux is infinity, as shown in here. I removed it to avoid the large difference. But I think I can add [1:] for analytical plotting to ensure consistency. 
ver-1b_comparison_flux

File: doc/content/verification_and_validation/ver-1gc.md (hunk 13 above)
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/ver-1gc.md:56]: There is nothing wrong in the python script. It just output the new RMSPE values different from the values on documentation. Maybe the gold files are updated in previous commits?

File: doc/content/verification_and_validation/ver-1fd.md (hunk 15 above)
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/ver-1fd.md:28]: The value is different with the one in documentation after rerunning the python script. So I update it with latest value.

File: doc/content/verification_and_validation/ver-1fc.md (hunk 16 above)
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/ver-1fc.md:43]: The value is different with the one in documentation after rerunning the python script. So I update it with latest value. Maybe I can rerun the input file with very fine mesh to lower the RMSPE.

File: test/tests/ver-1ja/tests (hunk 3 above)
Comment (simopier) [reply test/tests/ver-1ja/tests:16]: I don't think using the `heavy` tag is a good idea here because it's not really a heavy test (none of the ver-1ja tests - or several other cases - are heavy). 

This is a question for @cticenhour. Is there another tag we could/should use? Should the Apptainer tests be changed? 
He used the `heavy=true` tag in ver-1a because it was indeed plotting the results of a heavy test.

File: doc/content/verification_and_validation/ver-1fc.md (hunk 16 above)
Comment (simopier) [reply doc/content/verification_and_validation/ver-1fc.md:43]: No need to do that yet.

File: test/tests/ver-1b/comparison_ver-1b.py (hunk 11 above)
Comment (simopier) [reply test/tests/ver-1b/comparison_ver-1b.py:91]: Ok, I get it, it's good with your correction.

**Comment by simopier** [comment]
You need to list an issue number in one of your commit message:

```
##########################################################################
ERROR: Your patch does not contain a valid ticket reference! (i.e. #1234)
Merge branch 'doc_auto_figures' of https://github.com/lin-yang-ly/TMAP8 into test
ver1b_typos
Solve issue in test env and remove figures in documentation
Update documentation to generate figrues automatically - update V&V documentations - update V&V python script - update images
##########################################################################
```

File: test/tests/val-2b/comparison_val-2b.py [hunk 19]
```diff
@@ -4,13 +4,17 @@
...
 
+# Changes working directory to script directory (for consistent MooseDocs usage)
+os.chdir(os.path.dirname(__file__))
 
 fig = plt.figure(figsize=[6.5, 5.5])
 gs = gridspec.GridSpec(1, 1)
 ax = fig.add_subplot(gs[0])
 
-expt_data = pd.read_csv("./gold/experimental_data.csv")
+expt_data = pd.read_csv(os.path.join(git.Repo('.',search_parent_directories=True).working_tree_dir, "test/tests/val-2b/gold/experimental_data.csv"))
```
Comment (cticenhour) [inline test/tests/val-2b/comparison_val-2b.py:17]: The usage of `git.Repo` here is the root of the failure in Apptainer that necessitated use of `heavy` as a "workaround". This is because Apptainer installs a copy of TMAP8 and its tests, and then tests them. At that stage, it is no longer in a git repository, but on the regular apptainer file system. Hence, an `InvalidGitRepositoryError` is raised.

The use of `os.chdir(os.path.dirname(__file__))` on Line 11 of this file, which sets the working directory to the script directory, makes the directory references in the code more robust by setting the working directory to a standard location. So, all directory references can simply be relative to the working directory. You should just be able to say `./gold/experimental_data.csv` here when reading the CSV data. Why are you using `git.Repo` to do that? Was `chdir` not working correctly in your testing?

Comment (simopier) [reply test/tests/val-2b/comparison_val-2b.py:17]: Thank you so much @cticenhour!
My guess is that @lin-yang-ly was doing so because he used ver-1a as an example, and it was using:
```
tmap8_prediction = pd.read_csv(os.path.join(git.Repo('.',search_parent_directories=True).working_tree_dir, "test/tests/ver-1a/gold/ver-1a_csv.csv"))
```

@lin-yang-ly, could you use @cticenhour's recommendation and let us know if that works? And if it does, update ver-1a as well. 
Thank you!

Comment (cticenhour) [reply test/tests/val-2b/comparison_val-2b.py:17]: Yes, give it a try - I made that script when we weren't using Apptainer and installing TMAP8 on every PR.

Comment (lin-yang-ly) [reply test/tests/val-2b/comparison_val-2b.py:17]: Thank you, @cticenhour and @simopier! I will try to use the relative path based on the script directory.

Comment (lin-yang-ly) [reply test/tests/val-2b/comparison_val-2b.py:17]: It works with the updated relative path at #175! I will continue to update all cases in this PR.

Also, due to we call the python script from two directories (test folder and documentation folder), I use a "if" to distinguish the two working directories. Just let me know if you have better idea.
```
# Changes working directory to script directory (for consistent MooseDocs usage)
script_folder = os.path.dirname(__file__)
os.chdir(script_folder)

if "/TMAP8/doc/" in script_folder:     # if in documentation folder
    csv_folder = "../../../../test/tests/ver-1fa/gold/ver-1fa_csv_line_0011.csv"
else:                                  # if in test folder
    csv_folder = "./gold/ver-1fa_csv_line_0011.csv"
```

File: test/tests/ver-1b/tests [hunk 20]
```diff
@@ -39,6 +39,6 @@
       requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1b, modeling transient diffusion through a slab with a constant concentration boundary condition as the species source.'
       heavy = true
       prereq = ver-1b_heavy
```
Comment (simopier) [inline test/tests/ver-1b/tests:41]:
```suggestion
```

File: test/tests/ver-1c/tests [hunk 21]
```diff
@@ -20,6 +20,6 @@
     command = 'python3 comparison_ver-1c.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1c, modeling species permeation into an unloaded portion of a slab from a pre-loaded portion for both the TMAP4 and TMAP7 verification cases.'
     prereq = 'ver-1c_csvdiff'
```
Comment (simopier) [inline test/tests/ver-1c/tests:22]:
```suggestion
```

File: test/tests/ver-1d/tests [hunk 22]
```diff
@@ -60,6 +60,6 @@
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solutions of verification cases 1d, modeling a breakthrough problem where diffusion and trapping are the rate limiting processes.'
     heavy = true
     prereq = 'ver-1d_diffusion_limited_heavy ver-1d_trapping_limited_heavy'
```
Comment (simopier) [inline test/tests/ver-1d/tests:62]:
```suggestion
```

File: test/tests/ver-1dc/tests [hunk 23]
```diff
@@ -32,6 +32,6 @@
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solutions of verification cases 1dc, modeling a breakthrough problem of multiple traps.'
     heavy = true
     prereq = 'ver-1dc_limited_heavy'
```
Comment (simopier) [inline test/tests/ver-1dc/tests:34]:
```suggestion
```

File: test/tests/ver-1e/tests [hunk 24]
```diff
@@ -72,6 +72,6 @@
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1e, modeling transient diffusion through a composite slab with a constant concentration boundary condition as the species source for both the TMAP4 and TMAP7 verification cases.'
     heavy = true
     prereq = 'ver-1e_TMAP4_heavy ver-1e_TMAP7_heavy'
```
Comment (simopier) [inline test/tests/ver-1e/tests:74]:
```suggestion
```

File: test/tests/ver-1fa/tests [hunk 25]
```diff
@@ -21,6 +21,6 @@
     command = 'python3 comparison_ver-1fa.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1fa, to model heat conduction in a slab that has heat generation.'
     prereq = ver-1fa
```
Comment (simopier) [inline test/tests/ver-1fa/tests:23]:
```suggestion
```

File: test/tests/ver-1fc/tests [hunk 26]
```diff
@@ -37,6 +37,6 @@
     command = 'python3 comparison_ver-1fc.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution, ABAQUS data, and simulated solution of verification case 1fc, modeling conduction in a composite structure with constant surface temperatures.'
     prereq = 'ver-1fc_csvdiff_transient ver-1fc_csvdiff_steady_state'
```
Comment (simopier) [inline test/tests/ver-1fc/tests:39]:
```suggestion
```

File: test/tests/ver-1fd/tests [hunk 27]
```diff
@@ -20,7 +20,7 @@
     type = RunCommand
     command = 'python3 comparison_ver-1fd.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1fd, modeling convective heating.'
-    required_python_packages = 'csv matplotlib scipy numpy pandas os math'
+    required_python_packages = 'matplotlib scipy numpy pandas os'
     prereq = ver-1fd_csv
```
Comment (simopier) [inline test/tests/ver-1fd/tests:24]:
```suggestion
```

File: test/tests/ver-1g/tests [hunk 28]
```diff
@@ -51,6 +51,6 @@
     command = 'python3 comparison_ver-1g.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a chemical reaction between two species with same or different concentrations, using the initial conditions from both TMAP4 and TMAP7 cases.'
     prereq = 'binary_reaction_equal_concentrations binary_reaction_diff_concentrations_TMAP4 binary_reaction_diff_concentrations_TMAP7'
```
Comment (simopier) [inline test/tests/ver-1g/tests:53]:
```suggestion
```

File: test/tests/ver-1gc/tests [hunk 29]
```diff
@@ -22,6 +22,6 @@
     command = 'python3 comparison_ver-1gc.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a series of chemical reactions involving three species and calculate the concentrations of each species as a function of time and to generate CSV data for use in comparisons with the analytic solution over time.'
     prereq = 'ver-1gc_csv'
```
Comment (simopier) [inline test/tests/ver-1gc/tests:24]:
```suggestion
```

File: test/tests/ver-1ha/comparison_ver-1ha.py [hunk 30]
```diff
@@ -86,14 +89,14 @@ def get_analytical_solution(t_vect):
 # Root Mean Square Percentage Error calculations
 RMSE_C2 = np.linalg.norm(tmap8_sol_C2-C2)
 err_percent_C2 = RMSE_C2*100/np.mean(C2)
-ax.text(13, 1.95e20, '(C2) RMSPE = %.2f ' %
+ax.text(13, 2.25e20, '(C2) RMSPE = %.2f ' %
```
Comment (simopier) [inline test/tests/ver-1ha/comparison_ver-1ha.py:92]: Good catch!

File: test/tests/ver-1ha/tests [hunk 31]
```diff
@@ -13,6 +13,6 @@
       command = 'python3 comparison_ver-1ha.py'
       requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of a convective outflow problem involving three enclosures and calculate the pressure and concentration of the gas in the second and third enclosure.'
       prereq = 'ver-1ha_csv'
```
Comment (simopier) [inline test/tests/ver-1ha/tests:15]:
```suggestion
```

File: test/tests/ver-1ja/tests [hunk 32]
```diff
@@ -13,6 +13,6 @@
     command = 'python3 comparison_ver-1ja.py'
     requirement = 'The system shall be able to generate comparison plots between the analytical solution and simulated solution of verification case 1ja, which models decay of tritium and associated growth of helium in a diffusion segment.'
     prereq = ver-1ja_csvdiff
```
Comment (simopier) [inline test/tests/ver-1ja/tests:15]:
```suggestion
```

File: test/tests/val-2b/comparison_val-2b.py (hunk 19 above)
Comment (simopier) [reply test/tests/val-2b/comparison_val-2b.py]: That works for me, thank you!

**Review by simopier — CHANGES_REQUESTED** [review]
Awesome work on cleaning up the latest problems you were running into. 

As I described in one of the comments: now that we only use the gold files, the .csv file are always available to the python file to plot them, so no need to add the prereq and heavy flags. So please remove them on all the comparison RunCommand tests (I made suggestions, please make sure I didn't miss any).

**Comment by lin-yang-ly** [comment]
> Awesome work on cleaning up the latest problems you were running into.
> 
> As I described in one of the comments: now that we only use the gold files, the .csv file are always available to the python file to plot them, so no need to add the prereq and heavy flags. So please remove them on all the comparison RunCommand tests (I made suggestions, please make sure I didn't miss any).

Thanks for your comments, @simopier ! All the "heavy" and "prereq" flags in RunCommand are removed. The PR is good to move forward!

**Review by simopier — APPROVED** [review]
Thank you for this great improvement to the documentation!
